### ISSUE TITLE
Program Stopped Getting Data

### ISSUE THREAD

**Issue by landry314** [arminha/worldclockwidget#66]
The app just stopped getting data. I have no data for no cities even though everything is the same and all the settings are the same.

Anyone else having this problem?

**Comment by J-P-** [comment]
Yes. It says as much on the front page. It appears that the app is inactive.

**Comment by landry314** [comment]
but it was JUST updated... wth? :-(

**Comment by nospam2k** [comment]
Here is a fix (YahooWeatherService.class):

       // NEW URL http://query.yahooapis.com/v1/public/yql?q=select%20*%20from%20weather.forecast%20where%20woeid=2460286%20and%20u=%22c%22
        // String query = "w=" + woeid + "&u=c";
        String query = "q=" + "select * from weather.forecast where woeid=" + woeid + " and u=\"c\"";
        try {
            // java.net.URI uri = new URI("http", "weather.yahooapis.com", "/forecastrss", query, null);
            java.net.URI uri = new URI("http", "query.yahooapis.com", "/v1/public/yql", query, null);

**Comment by arminha** [comment]
In short: I disabled the weather function in the last release because it is broken and I don't have the time to fix it.

The weather function so far uses **one** API key for Open Weather Map compiled into the app. Their free account currently allows 60 calls per minute, which did not scale well with the growing number of users of the app.  Open Weather Map disabled my key temporarily after it exceeded the limits tenfold. That's why I disabled the function completely.

My preferred fix would be to allow users to add their own API key in the settings and only enable the weather function if they do. With a solution like this we could also bring back support for Yahoo weather as an alternative to Open Weather Map. Yahoo has even stricter limits for free accounts.

I myself will not implement a fix like that though, as I don't have the free time nor motivation to work on the app any more.
